src/BusinessLogic/Resources/js/TemplateService.js

```diff
@@ -49,87 +50,58 @@ var Packlink = window.Packlink || {};
          *
          * @return {NodeListOf<Element>}
          */
-        this.getComponentsByAttribute = function (attribute, element) {
+        this.getComponentsByAttribute = (attribute, element) => {
             let selector = '[' + attribute + ']';
 
-            if (typeof element === 'undefined') {
+            if (!element) {
                 return document.querySelectorAll(selector);
             }
 
             return element.querySelectorAll(selector);
         };
 
         /**
-         * Changes currently active page.
+         * Sets the content templates.
          *
-         * @param {string} template
-         * @param {string} [extensionPointIdentifier]
-         * @param {boolean} [clearExtensionPoint=true]
-         *
-         * @return {Element}
+         * @param {{}} configuration
          */
-        this.setTemplate = function (template, extensionPointIdentifier, clearExtensionPoint) {
-            if (typeof extensionPointIdentifier === 'undefined') {
-                extensionPointIdentifier = 'pl-content-extension-point';
-            }
-
-            if (typeof clearExtensionPoint === 'undefined') {
-                clearExtensionPoint = true;
-            }
-
-            let extensionPoint = this.getComponent(extensionPointIdentifier);
-
-            if (clearExtensionPoint) {
-                this.clearComponent(extensionPoint);
-            }
-
-            let templateElements = this.getTemplate(template);
-            while (templateElements && templateElements.length) {
-                extensionPoint.appendChild(templateElements[0]);
-            }
-
-            return extensionPoint;
+        this.setTemplates = configuration => {
+            templates = configuration;
         };
 
         /**
-         * Removes component's children.
+         * Gets the template with translated text.
          *
-         * @param {Element} component
+         * @param {string} templateId
+         *
+         * @return {string} HTML as string.
          */
-        this.clearComponent = function (component) {
-            while (component.firstChild) {
-                component.removeChild(component.firstChild);
-            }
-        };
+        this.getTemplate = templateId => Packlink.translationService.translateHtml(templates[templateId]);
```

Comment:
Codacy found an issue: ['Packlink' is not defined.](https://app.codacy.com/gh/packlink-dev/ecommerce_module_core/pullRequest?prid=5826400)